src/material/autocomplete/autocomplete.spec.ts

```diff
@@ -2559,7 +2573,7 @@ describe('MDC-based MatAutocomplete', () => {
 
       expect(input.value).toBe('Oregon');
       expect(stateCtrl.value).toEqual({code: 'OR', name: 'Oregon'});
-      expect(spy).toHaveBeenCalledTimes(1);
+      expect(spy).toHaveBeenCalledTimes(2);
```

Comment:
I don't quite follow why we're emitting an additional event here and the test below?

Reply:
This test and (the test below) start off by initializing the value to `CA`.

Previously, when the `OR` option was clicked, there was nothing to deselect, so it would only trigger one event: a select event on the `OR` option selection.

Now, the `CA` option is selected by setting the initial value, which means when the `OR` option is clicked, it triggers two events: a deselect event on the `CA` option and a select event on the `OR` option.

This happens when `_setValueAndClose` calls `_clearPreviousSelectedOption`, neither of which really changed, just the "initial conditions" are different, in a sense. I.e. I expect the count increasing by 2 whenever an option is clicked to be the typical behavior outside of the first time an option is selected, both before and after the changes in this PR.

Reply:
Should we add a comment here so there's no confusion about it in the future or change the initial test conditions?

Reply:
Added comments (`withContext` actually, since that seemed to be pretty prevalent) about what the expect should see. Can add more detail if needed but hopefully that should clue readers in enough about what's going on there.